packages/grafana-data/src/types/datasource.ts

```diff
@@ -535,19 +542,19 @@ export interface DataSourceSelectItem {
 /**
  * Options passed to the datasource.annotationQuery method. See docs/plugins/developing/datasource.md
  */
-export interface AnnotationQueryRequest<MoreOptions = {}> {
+export interface AnnotationQueryRequest<TAnno = {}> {
   range: TimeRange;
   rangeRaw: RawTimeRange;
+  interval: string;
+  intervalMs: number;
+  maxDataPoints?: number;
+  app: CoreApp | string;
```

Comment:
`app` is confusing here, first thing that came to my mind was - why is the app object passed to the request? Alternative would be `context` or `appContext`?

Reply:
This just copies what exists in the query request object -- I think added so that an explore query could have different behavior.

I added it so I could set a flag when in the annotation editor